modules/apps/oauth2-provider/oauth2-provider-service/src/main/java/com/liferay/oauth2/provider/service/impl/OAuth2ScopeGrantLocalServiceImpl.java

```diff
@@ -129,7 +129,14 @@ public Collection<OAuth2ScopeGrant> getOAuth2ScopeGrants(
 		List<OAuth2ScopeGrant> oAuth2ScopeGrants = new ArrayList<>();
 
 		oAuth2Authorizations.forEach(
```

Comment:
Avoid the 2 forEach in this plain looping case, they are anti-debugger and the lambdas adds extra call stack which is bad for performance. Do explicit for-loop as the code I pasted above.